checkstyle.xml

```diff
@@ -53,7 +53,12 @@
         <module name="TypeName"/>
         <module name="IllegalImport"/> <!-- defaults to sun.* packages -->
         <module name="IllegalImport">
-            <property name="illegalPkgs" value="clover, junit.framework, org.apache.commons.lang, org.slf4j, com.google.common.collect"/>
+            <property name="illegalPkgs" value="clover,
+                                                com.google.common.collect,
+                                                junit.framework,
+                                                org.apache.commons.lang,
+                                                org.apache.log4j,
+                                                org.slf4j"/>
```

Comment:
@dkjellin - Do you know why we banned this? It was added in #9 and I can see that `java.util.logging.Logger` is what we use everywhere else... but I don't understand why (and I couldn't find any reasoning in the PRs I found).

The Jenkins POM provides us with SLF4J and SLF4J-provided APIs for Log4j and JCL. So it seems like Jenkins is OK with plugins using SLF4J.

After seeing what torture we have to go through to use Java Logging (formatting strings with `Object` arrays, using `LOGGER.log(Level.WARN, "blah blah", e)` so that we can pass an exception with the message), I figure there must be a strong reason for avoiding a more modern API.

Reply:
Got an answer on Slack and added a comment about why.